routing/car_model.hpp

```diff
@@ -1,16 +1,29 @@
 #pragma once
-
 #include "vehicle_model.hpp"
 
+#include "std/shared_ptr.hpp"
+
 namespace routing
 {
 
 class CarModel : public VehicleModel
 {
+public:
```

Comment:
The problem is CarModelFactory creates CarModel with the help of  make_shared<CarModel>(..). So if I declare CarModelFactory as a friend of CarModel I still cannot use constructor in make_shared. That means cannot call make_shared<CarModel>().

There's exactly the same situation with PedestrianModel and BicycleModel.